### DISCUSSION TITLE
auto build not working well

### DISCUSSION THREAD

**Discussion by saintmalik** [codemagic-ci-cd/codemagic-docs#2473] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

Set automated build to trigger based on modifications on distribution/

workflows:
  rn-ios-dev:
    name: React Native iOS
    triggering:
      events:
        - push
      branch_patterns:
        - pattern: 'main'
          include: true
          source: true
    when:
      changeset:
        includes:
          - 'distribution/'

### Expected results

every commit with no modification to distribution/ build should be skipped

### Actual results

every commits every commit with no modification to distribution/ is running without skipping

**Comment by dtrdic** [comment]
hey @saintmalik just a note that after changeset setting is configured in codemagic.yaml, the subsequent build will be triggered regardless of the condition and only after that successful build, builds will be skipped according to the changeset condition.
So please run a successful build first, then try again. Please also share a build ID in the future as it's much easier for us to check the logs that way

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
